examples/ErrorHandling/HandlePartialFailures.cs

```diff
@@ -0,0 +1,166 @@
+            {
+                return "This code example demonstrates how to handle partial failures.";
+            }
+        }
+
+        /// <summary>
+        /// Runs the code example.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The customer ID for which the call is made.</param>
+        /// <param name="campaignId">Id of the campaign to which ad groups are added.</param>
+        public void Run(GoogleAdsClient client, long customerId, long campaignId)
+        {
+            // Get the AdGroupServiceClient.
+            AdGroupServiceClient adGroupService = client.GetService(Services.V1.AdGroupService);
+
+
+            // This ad group should be created successfully - assuming the campaign in the params
+            // exists.
+            AdGroup adgroup1 = new AdGroup()
+            {
+                Campaign = ResourceNames.Campaign(customerId, campaignId),
+                Name = "Valid AdGroup: " + ExampleUtilities.GetRandomString()
+            };
+
+            // This ad group will always fail - campaign ID 0 in resource names is never valid.
+            AdGroup adgroup2 = new AdGroup()
+            {
+                Campaign = ResourceNames.Campaign(customerId, 0),
+                Name = "Broken AdGroup: " + ExampleUtilities.GetRandomString()
+            };
+
+            // This ad group will always fail - duplicate ad group names are not allowed.
+            AdGroup adgroup3 = new AdGroup()
+            {
+                Campaign = ResourceNames.Campaign(customerId, 0),
+                Name = adgroup1.Name
+            };
+
+            // Create the operations.
+            AdGroupOperation op1 = new AdGroupOperation()
+            {
+                Create = adgroup1
+            };
+
+            AdGroupOperation op2 = new AdGroupOperation()
+            {
+                Create = adgroup2
+            };
+
+            AdGroupOperation op3 = new AdGroupOperation()
+            {
+                Create = adgroup3
+            };
+            
+            try
+            {
+                // Add the ad groups.
+                MutateAdGroupsResponse response =
+                    adGroupService.MutateAdGroups(customerId.ToString(),
+                        new AdGroupOperation[] { op1 }, true, false);
+
+                // Display the results.
+                if (response.PartialFailureError == null)
+                {
+                    Console.WriteLine("All operations completed successfully. No partial " +
+                        "failures to show.");
+                    return;
+                }
+                else
+                {
+                    // Finds the failed operations by looping through the results.
+                    int operationIndex = 0;
+                    foreach (MutateAdGroupResult result in response.Results)
+                    {
+                        if (result.IsEmpty())
```

Comment:
Might be worth calling out here that result.IsEmpty() implies partial failure occurred for that operation.

Reply:
Done